### PR TITLE
cli: use binary test validator

### PR DESCRIPTION
**Pull request by 2501babe**
for cli tests, instead of depending on and using the `solana-test-validator` crate, use the `solana-test-validator` binary we get from cli tools. currently #729 is unresolvable because this dep puts cli into hell

i would have liked to address #45 but using one validator for all tests a) requires them to be serial-ish still because it cant handle aggressive parallelism, and b) tests would need to be more extensively reworked around the fact that tests can cross epoch boundaries at any time. i dont think its worth it at present

### CODE REVIEW

File: clients/cli/tests/test.rs [hunk 1]
```diff
@@ -49,11 +51,108 @@ pub struct Env {
...
+        .stdout(Stdio::null())
+        .stderr(Stdio::null());
+    unsafe {
+        command.pre_exec(|| {
+            if libc::prctl(libc::PR_SET_PDEATHSIG, libc::SIGKILL) == -1 {
+                return Err(std::io::Error::last_os_error());
+            }
+            Ok(())
+        });
+    }
```
Comment (2501babe) [inline clients/cli/tests/test.rs:121]: under normal happy conditions we use `Drop` to ensure the validator is killed after its test ends, but this tells linux to kill -9 the validator if its parent dies. nothing drives me crazier than cli tools that leave behind heavy battery-killing processes that i dont notice for days (solana-verify used to do this with entire docker containers)
